pkg/container-collection/options.go

```diff
@@ -432,3 +438,280 @@ func WithLinuxNamespaceEnrichment() ContainerCollectionOption {
+			query, err := gojq.Parse(field)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse field %q: %w", field, err)
+			}
+
+			var outputs []interface{}
+			iter := query.Run(resourceJSON)
+			for {
+				value, ok := iter.Next()
+				if !ok {
+					break
+				}
+				if err, isErr := value.(error); isErr {
+					log.Errorf("failed to get field %q: %s", field, err)
+					continue
+				}
+				if value == nil {
+					continue
+				}
+				outputs = append(outputs, value)
+			}
+
+			return outputs, nil
+		}
+
+		getStringField := func(field string, resourceJSON interface{}) (string, error) {
+			values, err := getResourceField(field, resourceJSON)
+			if err != nil || len(values) == 0 {
+				return "", fmt.Errorf("failed to retrieve string field %q: %w", field, err)
+			}
+
+			// This function does not support array values
+			if len(values) != 1 {
+				return "", fmt.Errorf("unexpected number of matches for field %s: %d (expecting 1)",
+					field, len(values))
+			}
+
+			var strValues []string
+			for _, val := range values {
+				strVal, ok := val.(string)
+				if !ok {
+					log.Errorf("failed to convert to string field %s: %s", field, val)
+					continue
+				}
+				strValues = append(strValues, strVal)
+			}
+
+			return strValues[0], nil
+		}
+
+		getOwnerReference := func(resourceJSON interface{}) (*metav1.OwnerReference, error) {
+			values, err := getResourceField(".metadata.ownerReferences", resourceJSON)
+			if err != nil {
+				return nil, fmt.Errorf("failed to retrieve owner references: %w", err)
+			}
+
+			// No owner reference found
+			if len(values) == 0 {
+				return nil, nil
+			}
+
+			// TODO: How should we manage multiple references?
+			if len(values) != 1 {
+				log.Warnf("Taking only first owner reference and ignoring the rest: %v", values)
+			}
+
+			// Convert it into JSON to then use metav1.OwnerReference
+			var valueBytes []byte
+			valueBytes, err = json.Marshal(values[0])
+			if err != nil {
+				return nil, fmt.Errorf("failed to marshal owner references %v: %w", values[0], err)
+			}
+
+			ownerReferences := []metav1.OwnerReference{}
+			err = json.Unmarshal(valueBytes, &ownerReferences)
+			if err != nil {
+				return nil, fmt.Errorf("failed to unmarshal owner references %v: %w",
+					string(valueBytes), err)
+			}
+
+			return &ownerReferences[0], nil
+		}
+
+		// Future containers
+		cc.containerEnrichers = append(cc.containerEnrichers, func(containerDefinition *pb.ContainerDefinition) bool {
+			params := &ResourceParameters{
+				groupVersion: schema.GroupVersionResource{
+					Group:    "",
+					Version:  "v1",
+					Resource: "pods",
+				},
+				namespace: containerDefinition.Namespace,
+				listOpt: metav1.ListOptions{
+					FieldSelector: fields.OneTermEqualSelector("spec.nodeName", nodeName).String(),
+				},
+			}
+
+			pods, err := getResourceList(dynamicClient, params)
+			if err != nil {
+				log.Errorf("couldn't fetch pods: %s", err)
+				return true
+			}
```

Comment:
It's not needed to list the pods here. The container definition already must have the pod name and namespace, with that information you're able to fetch only the pod you're interested in.